### PR TITLE
Retrieving resources

### PR DESCRIPTION
**Pull request by cataraul**

### CODE REVIEW

**Review by MariaCrucin — APPROVED** [review]

**Review by AlinCuc — APPROVED** [review]

File: TweetBook/Contract/V1/Requests/CreatePostRequest.cs [hunk 1]
```diff
@@ -2,6 +2,6 @@
 {
     public class CreatePostRequest
     {
-        public string Id { get; set; }
+        public Guid Id { get; set; }
```
Comment (tatarflavia) [inline TweetBook/Contract/V1/Requests/CreatePostRequest.cs:5]: It's better to work directly with the type you are using, good change.

File: TweetBook/Program.cs [hunk 2]
```diff
@@ -1,7 +1,11 @@
+﻿using TweetBook.Services;
+
 var builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
 
+builder.Services.AddSingleton<IPostService, PostService>();
+
 builder.Services.AddControllers();
 // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
```
Comment (tatarflavia) [inline TweetBook/Program.cs]: Maybe you can delete all the non-relevant to the project comments like this one. Besides this, you can keep the ones which help you understand your code, although in real life we put comments only in exceptional situations where not any developer would understand at first glance the code.

File: TweetBook/Controllers/V1/PostsController.cs [hunk 3]
```diff
@@ -3,25 +3,45 @@
...
         public IActionResult GetAll()
         {
+
+            return Ok(_postService.GetPosts());
+        }
+
+        [HttpGet(ApiRoutes.Posts.Get)]
+        public IActionResult Get([FromRoute]Guid postId)
+        {
+
```
Comment (tatarflavia) [inline TweetBook/Controllers/V1/PostsController.cs:29]: Please keep in mind that after the start parenthesis for a method/class we don't leave any blank rows.

File: TweetBook/Controllers/V1/PostsController.cs [hunk 4]
```diff
@@ -30,16 +50,15 @@ public IActionResult Create([FromBody] CreatePostRequest postRequest)
 
             var post = new Post { Id = postRequest.Id };
 
-            if(string.IsNullOrEmpty(post.Id))
-                post.Id = Guid.NewGuid().ToString();
-            
-            _posts.Add(post);
+            if (post.Id != Guid.Empty)
```
Comment (tatarflavia) [inline TweetBook/Controllers/V1/PostsController.cs]: Here shouldn't we want to have a new Id only if we did not get one in the first place?

File: TweetBook/Controllers/V1/PostsController.cs [hunk 5]
```diff
@@ -30,16 +50,15 @@ public IActionResult Create([FromBody] CreatePostRequest postRequest)
...
+                post.Id = Guid.NewGuid();
+
+            _postService.GetPosts().Add(post);
```
Comment (tatarflavia) [inline TweetBook/Controllers/V1/PostsController.cs:47]: Please create a method for adding a post in the service.

File: TweetBook/Controllers/V1/PostsController.cs [hunk 6]
```diff
@@ -30,16 +50,15 @@ public IActionResult Create([FromBody] CreatePostRequest postRequest)
...
 
             var baseUrl = $"{HttpContext.Request.Scheme}://{HttpContext.Request.Host.ToUriComponent()}";
```
Comment (tatarflavia) [inline TweetBook/Controllers/V1/PostsController.cs:50]: Please erase the blank row.

File: TweetBook/Services/PostService.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        public List<Post> GetPosts()
+        {
+            return _posts;
+        }
+
+        public Post GetPostById(Guid postId)
+        {
+            return _posts.SingleOrDefault(x => x.Id == postId);
+        }
+
```
Comment (tatarflavia) [inline TweetBook/Services/PostService.cs]: please delete the blank rows between the 2 parentheses.

File: TweetBook/Services/IPostService.cs [hunk 8]
```diff
@@ -0,0 +1,11 @@
+﻿using TweetBook.Domain;
+
+namespace TweetBook.Services
+{
+    public interface IPostService
+    {
+        List<Post> GetPosts();
```
Comment (tatarflavia) [inline TweetBook/Services/IPostService.cs]: Please also add the access modifier,  as a good practice

Comment (tatarflavia) [reply TweetBook/Services/IPostService.cs]: For both methods

File: TweetBook/Program.cs [hunk 9]
```diff
@@ -1,7 +1,11 @@
+﻿using TweetBook.Services;
+
 var builder = WebApplication.CreateBuilder(args);
```
Comment (tatarflavia) [inline TweetBook/Program.cs:3]: Is this how the tutorial did it in the Program.cs? What about the Startup file, is it empty?

Comment (tatarflavia) [reply TweetBook/Program.cs:3]: In .Net 6 we don't have the startup file, my bad. Maybe in the future we will organise better this file, but for now this conversation is resolved.

**Review by tatarflavia — APPROVED** [review]

**Pull request merged**